sys/tlk_device.txt

```diff
@@ -0,0 +1,63 @@
+# Copyright 2016 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+# Open Trusted Execution driver for /dev/tlk_device
+# Reference source code:
+# https://android.googlesource.com/kernel/tegra/+/android-tegra-dragon-3.18-marshmallow-dr-dragon/security/tlk_driver/ote_protocol.h
+
+include <linux/ioctl.h>
+include <linux/types.h>
+include <security/tlk_driver/ote_protocol.h>
+
+syz_open_dev$tlk_device(dev strconst["/dev/tlk_device"], id const[0], flags flags[open_flags]) fd[tlk_device]
+
+ioctl$TE_IOCTL_OPEN_CLIENT_SESSION(fd fd[tlk_device], cmd const[TE_IOCTL_OPEN_CLIENT_SESSION], arg ptr[inout, te_cmd])
+ioctl$TE_IOCTL_CLOSE_CLIENT_SESSION(fd fd[tlk_device], cmd const[TE_IOCTL_CLOSE_CLIENT_SESSION], arg ptr[inout, te_cmd])
+ioctl$TE_IOCTL_LAUNCH_OPERATION(fd fd[tlk_device], cmd const[TE_IOCTL_LAUNCH_OPERATION], arg ptr[inout, te_cmd])
+ioctl$TE_IOCTL_SS_CMD(fd fd[tlk_device], cmd const[TE_IOCTL_SS_CMD], arg int32)
+
+
+te_cmd [
+	opensession	te_opensession
+	closesession	te_closesession
+	launchop	te_launchop
+]
+
+te_service_id {
+	time_low	int32
+	time_mid	int16
+	time_hi_and_version	int16
+	clock_seq_and_node	array[int8, 8]
+}
+
+te_opensession {
+        dest_uuid   te_service_id
+        operation   te_operation
+        answer      ptr[out, te_answer]
+}
+
+te_closesession {
+        session_id  int32
+        answer      ptr[in, te_answer]
+}
+
+te_answer {
+        result         int32
+        session_id     int32
```

Comment:
Add a TODO here that session_id should be represented by a separate resource type.
It's a bit tricky now as it will require to touch source files. So we can fit it when we simplify addition of new resources.